### PR TITLE
Added rule for final in classes

### PR DESCRIPTION
**Pull request by Haud**
@ThibaultKlein @paulmiard

### CODE REVIEW

**Comment by ThibaultKlein** [comment]
:+1:

**Comment by Haud** [comment]
? Care to expand?  I think this is a valuable rule and given that functions and classes can be final, we should absolutely enforce its usage in order to prevent bugs due to subclassing APIs that were never meant to be subclassed, shadowed, or overriden.

**Comment by ThibaultKlein** [comment]
LOL sorry I meant to put +1 @Haud

**Comment by Haud** [comment]
HAHAHAHA I was like "well that was aggressive, I thought it was a good rule!"

**Comment by Haud** [comment]
@ThibaultKlein Added an example. Can you just read through again and verify if it's good? Also still waiting on you @paulmiard

File: README.md [hunk 1]
```diff
@@ -425,4 +426,39 @@ In cases where the enum's type name can be omitted, do so:
...
+internal final class Object { // Not built for inheritance.
+
+}
+
+internal class BaseClass { // Purposefully utilizable as a base class
+	
+	func doSomething () {
+	}
+	
+	final func update() { // Properly marked as final so subclasses cannot override
```
Comment (ThibaultKlein) [inline README.md]: Can we move the comments above the functions?

**Comment by Haud** [comment]
@ThibaultKlein done.

File: README.md [hunk 2]
```diff
@@ -425,4 +426,42 @@ In cases where the enum's type name can be omitted, do so:
 let state = State.Open
 
 if state == .Closed { ... // Prefer .Closed instead of State.Closed
-```
\ No newline at end of file
+```
+
+
+### Use of `Final` ###
```
Comment (ThibaultKlein) [inline README.md]: should be `final`

**Comment by ThibaultKlein** [comment]
:+1:

**Pull request merged**